apps/speed.c

```diff
@@ -1776,6 +2145,24 @@ int speed_main(int argc, char **argv)
             goto end;
         }
     }
+    if (kems_algs_len > 0) {
+        int maxcnt = 0;
+        for (i = 0; i < kems_algs_len; i++)
+            if (maxcnt < kems_doit[i]) maxcnt = kems_doit[i];
```

Comment:
you may factorize these lines like this :  
```
int get_max(const int doit[], size_t algs_len) {
    int i, maxcnt = 0;

    for (i = 0; i < algs_len; ++i++)
        if (maxcnt < doit[i]) maxcnt = doit[i];
    return maxcnt;
}
```
And them call : 
`int maxcnt = get_max(kems_doit, kems_algs_len);`

Reply:
Thanks for the suggestion. So done in https://github.com/openssl/openssl/pull/19968/commits/d564ef96bab3c26de9652f743a1444f8f3015540.